### PR TITLE
feat: [GREP] - Usage - Filter available values on depended properties

### PR DESCRIPTION
**Pull request by gabrielfs7**
https://oat-sa.atlassian.net/browse/ADF-437
and
https://oat-sa.atlassian.net/browse/ADF-691

TODO: 

- [x] Add `parentListValues` so FE can filter only the properties available for that list.
- [x] When loading the page, bring only the child values allowed based on the parent value.
- [x] FE integration with the form.
- [x] Fix / create unit tests (Depends on other PR)

How to test:

Basically a new parameter `parentListValues[]` needs to be provided with the request.

```
curl --location -g --request GET 'https://test-tao-deploy.docker.localhost/tao/PropertyValues/get?propertyUri=https_2_test-tao-deploy_0_docker_0_localhost_1_ontologies_1_tao_0_rdf_3_i613ef9687600a9458b74958e40e4c4&subject=+++&_=1631536990809&parentListValues[]=http_2_test_0_test_0_test_1_PARENT-CODE-1&parentListValues[]=http_2_test_0_test_0_test_1_PARENT-CODE-2&parentListValues[]=http_2_test_0_test_0_test_1_PARENT-CODE-3' \
--header 'pragma: no-cache' \
--header 'cache-control: no-cache' \
--header 'accept: application/json, text/javascript, */*; q=0.01' \
--header 'x-requested-with: XMLHttpRequest' \
--header 'cookie: tao=xxxxxxxxx;'
```

### CODE REVIEW

File: models/classes/Lists/Presentation/Web/RequestHandler/ValueCollectionSearchRequestHandler.php [hunk 1]
```diff
@@ -32,12 +32,14 @@
 use oat\tao\model\service\InjectionAwareService;
 use Psr\Http\Message\ServerRequestInterface;
 use tao_helpers_Uri as Id;
+use function foo\func;
```
Comment (boajer) [inline models/classes/Lists/Presentation/Web/RequestHandler/ValueCollectionSearchRequestHandler.php:35]: ?

File: views/js/form/post-render-props.js [hunk 2]
```diff
@@ -310,7 +310,8 @@ define([
         }
         if (!$properties.length) {
             if ($container.children('[name="tao.forms.instance"]').length) {
-                secondaryProps.move($container);
+                secondaryProps.init($container);
+                // secondaryProps.move($container);
```
Comment (habemuscode) [inline views/js/form/post-render-props.js:314]: Remove comment

**Review by habemuscode — APPROVED** [review]
FE LGTM

File: views/js/form/post-render-props.js (hunk 2 above)
Comment (gabrielfs7) [reply views/js/form/post-render-props.js:314]: @clagosarias should we remove this?

Comment (clagosarias) [reply views/js/form/post-render-props.js:314]: yes, will do now!

File: models/classes/Lists/Presentation/Web/RequestHandler/ValueCollectionSearchRequestHandler.php (hunk 1 above)
Comment (gabrielfs7) [reply models/classes/Lists/Presentation/Web/RequestHandler/ValueCollectionSearchRequestHandler.php]: removed!

**Review by shpran — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by chinnu-focaloid — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Comment by AntonNeviarouski** [comment]
QA verified

**Pull request merged**